src/plugins/typescript/visitors.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2017-present Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain
+ * a copy of the license in the LICENSE and PATENTS files in the root of this
+ * repository.
+ */
+
+import { relative } from "path";
+import {
+    ContainerReflection,
+    DeclarationReflection,
+    ParameterReflection,
+    Reflection,
+    ReflectionKind,
+    SignatureReflection,
+} from "typedoc";
+import { Comment } from "typedoc/dist/lib/models/comments/comment";
+import { DefaultValueContainer } from "typedoc/dist/lib/models/reflections/abstract";
+import {
+    ITsClass,
+    ITsInterface,
+    ITsMethod,
+    ITsMethodSignature,
+    ITsParameter,
+    ITsProperty,
+    Kind,
+} from "../../client/typescript";
+import { ICompiler } from "../plugin";
+import { resolveSignature, resolveTypeString } from "./typestring";
+
+export type Renderer = ICompiler["renderBlock"];
```

Comment:
passing the `renderer` down was the quickest solution i had to the `documentation: IBlock` problem... deep inside these visitors, we suddenly need to render some documentation. typically, we're rendering at leaf nodes, but here i'm forced to pass it all the way down.

✋ help!